From 66149c4212a3bf124807843f3cdaac1ad4e078e7 Mon Sep 17 00:00:00 2001
From: Tor Lillqvist <tml@collabora.com>
Date: Fri, 20 Dec 2013 14:56:33 +0200
Subject: [PATCH] Prevent occasional crash

I came across the crash by accident when checking how it works to have
multiple windows showing the same document. Pasting in lots of text
and then quickly telling LO to exit with Cmd-Q caused a crash here as
GetPostItMgr() returned NULL.

Change-Id: Ib9e636df0832a679a1d81fa3856ea0a7105a69c3
(cherry picked from commit 0eb1ef39084a978e8c2bec977ebabf6708f0c073)
---
 sw/source/ui/misc/swruler.cxx | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sw/source/ui/misc/swruler.cxx b/sw/source/ui/misc/swruler.cxx
index fa396d0..b1035c9 100644
--- a/sw/source/ui/misc/swruler.cxx
+++ b/sw/source/ui/misc/swruler.cxx
@@ -56,7 +56,8 @@ void SwCommentRuler::Paint( const Rectangle& rRect )
 {
     SvxRuler::Paint( rRect );
     // Don't draw if there is not any note
-    if ( mpViewShell->GetPostItMgr()->HasNotes() )
+    if ( mpViewShell->GetPostItMgr()
+         && mpViewShell->GetPostItMgr()->HasNotes() )
         DrawCommentControl();
 }
 
-- 
1.9.3

